File: src/main/java/com/tim/spring/entity/Hero.java
package com.tim.spring.entity;

import org.springframework.data.jpa.domain.AbstractPersistable;

import javax.persistence.Column;
import javax.persistence.Entity;

@Entity
public class Hero extends AbstractPersistable<Long> {

    @Column(nullable = false, unique = true)
    private String name;

    public String getName() {
        return name;
    }

    public void setName(final String name) {
        this.name = name;
    }
}

File: src/main/kotlin/com/tim/spring/entity/Hero.kt
package com.tim.spring.entity

import org.springframework.data.jpa.domain.AbstractAuditable
import java.io.Serializable
import javax.persistence.*

@Entity
data class Hero(
        @Id
        var id: Long,

        @Column(nullable = false)
        var name: String

) : AbstractAuditable<User, Long>()